web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -124,17 +126,29 @@ export class MapComponent implements AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
+    this.initMap();
+  }
+
+  ngOnChanges() {
+    this.initMap();
```

Comment:
If we call `initMap()` multiple times, we'll accumulate one subscription for each change. Instead, there are two approaches you could take:

1. Do what you're doing here, but also dispose of the previous subscription before creating a new one, or
2. Create an Observable, selectedJobId$, which you combine with the other streams in `initMap()`.

Wdyt?

Reply:
Agreed, took option 2.